java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture2/tasks/_2/Number2.java

```diff
@@ -0,0 +1,164 @@
+        Scanner scanner = new Scanner(System.in);
+        boolean exit = false;
+        System.out.println("HELLO AND WELCOME!. Please input width, height, x, y coordinates and Direction");
+        RobotGame robotGame = new RobotGame(scanner.nextInt(), scanner.nextInt(), scanner.nextInt(), scanner.nextInt(), Direction.SOUTH);
+        System.out.println("Your room size is:");
+        System.out.println(robotGame.getRoomSize() + "\n");
+        System.out.println("Your coordinates are:");
+        System.out.println(robotGame.getCoordinates() + "\n");
+        System.out.println("Set direction where your robot is going. Default direction is SOUTH");
+        System.out.println("0. SOUTH,\n1. NORTH,\n2. WEST,\n3. EAST");
+        Direction direction = Direction.values()[Integer.parseInt(scanner.next())];
+        System.out.println(direction.toString() + "\n");
+
+        System.out.println("1.Move Forward.\n2.Change Direction;\n3.Get coordinates;\n4.Get directions;\n5.Exit");
+
+        while (!exit) {
+            switch (scanner.nextInt()) {
+                case 1 -> {
+                    System.out.println("Type N steps");
+                    int steps = scanner.nextInt();
+                    robotGame.moveForward(steps);
+                }
+                case 2 -> {
+                    System.out.println("1.Turn Left;\n2.Turn Right;");
+                    if (scanner.nextInt() == 1) {
+                        robotGame.turnLeft();
+                        System.out.println("Turned left");
+
+                    } else {
+                        robotGame.turnRight();
+                        System.out.println("Turned right");
+                    }
+                }
+                case 3 -> System.out.println(robotGame.getCoordinates());
+                case 4 -> robotGame.getDirection();
+                case 5 -> {
+                    System.out.println("Shutting down");
+                    exit = true;
+                }
+            }
+        }
+    }
+}
+
+enum Direction {
+    SOUTH,
+    NORTH,
+    WEST,
+    EAST
+}
+
+class RobotGame {
+
+    private final int width;
+    private final int height;
+    private int x;
+    private int y;
+    Direction direction;
+
+    public RobotGame(int width, int height, int x, int y, Direction direction) {
+        this.width = width;
+        this.height = height;
+        this.x = x;
+        this.y = y;
+        this.direction = direction;
+    }
+
+    public String getRoomSize() {
+        return "Width: " + width + " " + "Height: " + height;
+    }
+
+    public String getCoordinates() {
+        return "X: " + x + " " + "Y: " + y;
+    }
+
+    public void getDirection() {
+        System.out.println(direction);
+    }
+
+    public void moveForward(int n) {
+        if (n > height || n > width) {
+            System.out.println("Amount of steps are greater than a width or height of a room");
+        }
+        else if ((n + y) > height || (n + x) > width) {
```

Comment:
this condition will block robot when it's located near the SOUTH or EAST borders, and try to move either to NORTH or WEST:

n = 3, y = 4, height = 5, direction = WEST: (3 + 4) > 5 => true, even though it's a legit move